src/core/handlers/RequestHandler.ts

```diff
@@ -19,6 +19,14 @@ export type DefaultBodyType =
   | null
   | undefined
 
+export type JsonBodyType =
+  | Record<string, any>
```

Comment:
nit: do we want to be slightly more prescriptive than `Record<string, any>` here, `Record<string, JsonBodyType>` might be the most correct option?

also I think we'd need to account for `Array<JsonBodyType>` too? (and potentially their readonly varieties?)

Reply:
Surprisingly, TypeScript is permissive about this and arrays [extend](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAQwhApgBygeQEYBW60AzjmADYCeAFAG7IUjoBciASsXAE4AmAPCSjcYYAOYAaFGCoA+AJSIA3gF8AUGtQZs+IqXLUaq+ZrRZchYlDKVaAbQC6Jred1WbhuwHJGJOF6kvYFF0LycgA) `Record<string, any>` quite well. It's the `any` part that does the trick.

As to narrowing down the type from `any` to a recursive `JsonBodyType`, I'm not so sure about it. It sounds like the right thing to do but there's always corner cases people run into, and it's hard to predict what will happen if we make this body type _that_ strict.